src/transformStateWithClones.js

```diff
@@ -7,7 +7,40 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (type) {
+      case 'addProperties':
+        currentState = {
+          ...currentState,
+          ...extraData,
+        };
+        break;
+
+      case 'removeProperties':
+        keysToRemove.forEach(key => {
+          if (Object.hasOwnProperty.call(currentState, key)) {
```

Comment:
try it like this, hope it will work:
```suggestion
          if (Object.hasOwnProperty(key)) {
```